src/pages/forgotPassword/ResetPassword.tsx

```diff
@@ -0,0 +1,65 @@
+import { FunctionComponent, useState } from "react";
+import { FormikValues, useFormikContext } from "formik";
+import { Box, Stack, useTheme } from "@mui/material";
+import { Button, Typography } from "elements";
+import { PasswordInputField } from "components";
+
+const ResetPassword: FunctionComponent = () => {
+  const theme = useTheme();
+  const [maskPassword, setMaskPassword] = useState(true);
+  const {
+    isValid,
+    values: { confirmPassword, newPassword },
+  } = useFormikContext<FormikValues>();
+
+  const togglePasswordMask = () => {
+    setMaskPassword(!maskPassword);
+  };
+
+  const showEndAdornment = !!(newPassword || confirmPassword);
```

Comment:
This is just understanding for myself, I see that everything works correctly and the mask icon appears once a change occurs in one of the input text fields, but this is a `const` and I don't see anything in `PasswordInputField.tsx` that may change this value to trigger the `endAdornment`.

Reply:
Yea I do understand that portion I guess its the connection to this in `PassworInputField.tsx`:

```
endAdornment={
        showEndAdornment ? (
          <IconButton
            aria-label="Change password field visible status"
            onClick={ handlePressEndAdornment || togglePasswordMask }
          >
            <PasswordIcon sx={ { color: theme.colors.white } } />
          </IconButton>
        ) : undefined
      }
```
Where if showEndAdornment is set to false on empty strings and passed in, how does it update to become true, is it just wiffing my head as a basic react Component process, or is it a Formik trick in updating input fields and components